app/src/server/db/mongodb.ts

```diff
@@ -0,0 +1,33 @@
+import { MongoClient } from 'mongodb';
+
+if (!process.env.MONGODB_URI) {
+  throw new Error('Invalid/Missing environment variable: "MONGODB_URI"');
+}
+
+const uri = process.env.MONGODB_URI;
+const options = {};
+
+let client;
+let clientPromise: Promise<MongoClient>;
+
+if (process.env.NODE_ENV === 'development') {
+  // In development mode, use a global variable so that the value
+  // is preserved across module reloads caused by HMR (Hot Module Replacement).
+  const globalWithMongo = global as typeof globalThis & {
```

Comment:
Also I am a little confused on what this actually does?

Reply:
This ensures that the connection can just be re-used when the hot module replacement (fast hot reloading) happens. It makes sure that when using `npm run dev`, you aren't spamming connections to mongo every time your code updates.

In the actual built production code of the app, this code won't get hit and it will just use one connection since the app won't be continuously restarted

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis